### PR TITLE
feat(Request): allow aborting intercepted requests with custom reasons

### PR DESCRIPTION
**Pull request by aslushnikov**
This patch adds optional parameter to the `request.abort()` method
that specifies abort reason.

References #1020.

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -1178,6 +1178,17 @@ describe('Page', function() {
       expect(response.ok).toBe(true);
```
Comment (JoelEinbinder) [inline test/test.js:1178]: Can we check the error text in this test? just to be safe

**Review by JoelEinbinder — APPROVED** [review]

File: docs/api.md [hunk 2]
```diff
@@ -1831,7 +1831,22 @@ If request fails at some point, then instead of 'requestfinished' event (and pos
 
 If request gets a 'redirect' response, the request is successfully finished with the 'requestfinished' event, and a new request is  issued to a redirected url.
 
-#### request.abort()
+#### request.abort(errorCode)
```
Comment (JoelEinbinder) [inline docs/api.md]: [errorCode], because it is optional

File: test/test.js (hunk 1 above)
Comment (aslushnikov) [reply test/test.js:1178]: Done.

File: docs/api.md (hunk 2 above)
Comment (aslushnikov) [reply docs/api.md]: Done.

**Pull request merged**